File: DesktopApp/Utilities/Utilities.cs
using System.Security.Cryptography;
using System.Text;

namespace ProfessorPerformanceEvaluation.Utilities
{
    public static class Utilities
    {
        public static string ComputeSHA256Hash(string password)
        {
            using (SHA256 sha256Hash = SHA256.Create())
            {
                byte[] bytes = sha256Hash.ComputeHash(Encoding.UTF8.GetBytes(password));
                var stringBuilder = new StringBuilder();
                for (int bit = 0; bit < (bytes.Length); bit++)
                {
                    stringBuilder.Append(bytes[bit].ToString("x2"));
                }
                return stringBuilder.ToString();
            }
        }
    }
}

File: MobileApp/app/src/main/java/com/example/professorperformanceevaluation/utilities/Utilities.java
package com.example.professorperformanceevaluation.utilities;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class Utilities {

    public static String computeSHA256Hash(String password) {
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte[] bytes = digest.digest(password.getBytes(StandardCharsets.UTF_8));
            StringBuilder stringBuilder = new StringBuilder();
            for (byte item : bytes) {
                String hex = String.format("%02x", item);
                stringBuilder.append(hex);
            }
            return stringBuilder.toString();
        } catch (NoSuchAlgorithmException e) {
            return null;
        }
    }

    public static String formatterDate(Date fecha) {
        DateFormat formatoFechaEntrada = new SimpleDateFormat("yyyy-MM-dd");
        DateFormat formatoFechaSalida = new SimpleDateFormat("dd/MM/yyyy");

            return formatoFechaSalida.format(fecha);
    }

}